modules/tinymce/src/plugins/table/main/ts/actions/TableActions.ts

```diff
@@ -19,12 +19,14 @@ import * as TableSize from '../queries/TableSize';
 import { ephemera } from '../selection/Ephemera';
 import { getCellsFromSelection, getRowsFromSelection } from '../selection/TableSelection';
 
-type TableAction<T> = (table: SugarElement<HTMLTableElement>, target: T) => Optional<Range>;
+type TableAction<T> = (table: SugarElement<HTMLTableElement>, target: T) => Optional<TableActionResult>;
+export type TableActionResult = { rng: Range; effect: TableEventData };
 export type SimpleTableAction = (editor: Editor, args: Record<string, any>) => void;
 export type CombinedTargetsTableAction = TableAction<RunOperation.CombinedTargets>;
 export type PasteTableAction = TableAction<RunOperation.TargetPaste>;
 export type AdvancedPasteTableAction = TableAction<RunOperation.TargetPasteRows>;
 export type ElementTableAction = TableAction<RunOperation.TargetElement>;
+export type TableEventData = { structure: boolean; style: boolean };
```

Comment:
This would be better in `Events` as that's it's source location.

- [ ] Move to events and make it an interface please